### PR TITLE
Force tracking to diesel 2.1.6, to follow along with omicron

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Issue:

Pointing at the current `main` branch of async-bb8-diesel (which uses diesel 2.2.*), as it does, causes the build of this lib to use diesel 2.2.* as is, which throws this error on the impl of `Connection`:

```
error[E0046]: not all trait items implemented, missing: `instrumentation`, `set_instrumentation`
   --> /home/zeeshan/.cargo/git/checkouts/diesel-dtrace-b9b9a29f91f682b9/62ef5ca/src/lib.rs:117:1
    |
117 | / impl<C> Connection for DTraceConnection<C>
118 | | where
119 | |     C: Connection<TransactionManager = AnsiTransactionManager>,
120 | |     C::Backend: Default,
121 | |     <C::Backend as Backend>::QueryBuilder: Default,
    | |___________________________________________________^ missing `instrumentation`, `set_instrumentation` in implementation
    |
    = help: implement the missing item: `fn instrumentation(&mut self) -> &mut (dyn Instrumentation + 'static) { todo!() }`
    = help: implement the missing item: `fn set_instrumentation<impl Instrumentation>(&mut self, _: impl Instrumentation) where impl In
strumentation: Instrumentation { todo!() }`
```

Updating this crate's diesel version to 2.2 will cause issues with omicron, as omicron is set to 2.1.6, before the change took shape upstream in diesel. 

Note:
  * async-bb8-diesel without the rev will use 2.2.* diesel dep, adding new trait impl fns

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,15 +7,16 @@ license = "Apache-2.0"
...
 [dependencies]
-diesel = { version = "2.1.0", features = [ "r2d2", "i-implement-a-third-party-backend-and-opt-into-breaking-changes" ] }
+diesel = { version = "=2.1.6", features = [ "r2d2", "i-implement-a-third-party-backend-and-opt-into-breaking-changes" ] }
 serde = "1"
 usdt = "0.5"
 uuid = { version = ">=0.8.0, <2.0.0", features = [ "v4", "serde" ] }
 
 [dev-dependencies]
-async-bb8-diesel = { git = "https://github.com/oxidecomputer/async-bb8-diesel" }
+## match omicron
```
Comment (ahl) [inline Cargo.toml:16]: why is omicron pinned to a rev here? how do we envision advancing everyone forward?

File: Cargo.toml [hunk 2]
```diff
@@ -7,15 +7,16 @@ license = "Apache-2.0"
 repository = "https://github.com/oxidecomputer/diesel-dtrace.git"
 
 [dependencies]
-diesel = { version = "2.1.0", features = [ "r2d2", "i-implement-a-third-party-backend-and-opt-into-breaking-changes" ] }
+diesel = { version = "=2.1.6", features = [ "r2d2", "i-implement-a-third-party-backend-and-opt-into-breaking-changes" ] }
```
Comment (ahl) [inline Cargo.toml:10]: why is this necessary?

**Review by ahl — COMMENTED** [review]
I'm not clear on the problem we're solving. This seems to sediment the stalled update process in omicron rather than restarting it.

File: Cargo.toml (hunk 2 above)
Comment (zeeshanlakhani) [reply Cargo.toml:10]: @ahl the issue is that async-bb8-diesel off `main` will make Cargo.lock rely on diesel 2.2, which will break this building this dep b/c the impl on `Connection` (https://github.com/oxidecomputer/diesel-dtrace/blob/main/src/lib.rs#L117) has new functions in 2.2. 

I attempted to just set async-bb8 to the rev on omicron, but without the `=2.1.6`, I was still having it float to 2.2 deps.

File: Cargo.toml (hunk 1 above)
Comment (zeeshanlakhani) [reply Cargo.toml:16]: @ahl I'm not sure why omicron is pinned to a rev here. I'm going off what's in `main`, which hits at this: https://github.com/oxidecomputer/omicron/commit/b3d641a2dd3bb3f3dd68a413ddd670e3dbcc0b5c.

Comment (zeeshanlakhani) [reply Cargo.toml:16]: actually, the culprit may be: https://github.com/oxidecomputer/async-bb8-diesel/blob/master/Cargo.toml#L26 which has diesel 2.2.
